fluent-bundle/src/resolve.rs

```diff
@@ -13,271 +13,294 @@ use std::hash::{Hash, Hasher};
 
 use super::bundle::FluentBundle;
 use super::entry::GetEntry;
+use super::types::DisplayableNode;
 use super::types::FluentValue;
 use fluent_syntax::ast;
 use fluent_syntax::unicode::unescape_unicode;
 
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Clone)]
 pub enum ResolverError {
-    None,
+    Reference(String),
+    MissingDefault,
+    Argument(String),
     Value,
     Cyclic,
 }
 
 /// State for a single `ResolveValue::to_value` call.
-pub struct Env<'env> {
+pub struct Scope<'scope> {
```

Comment:
This lifetime is the same as the one on the bundle, right?

We should give it a uniform name across the codebase and name it, perhaps something like `'bundle`?

Typically lifetimes are named the same as the source of their data, in this case it seems to be "computed Fluent information stored in the bundle". Unless I'm misreading it and `'scope` is a shorter lifetime.

It's also useful to document where the data behind the lifetime comes from so that it's easier to reason about this.